drivers/serial/uart_stm32.c

```diff
@@ -525,21 +525,25 @@ static void uart_stm32_poll_out(const struct device *dev,
 					unsigned char c)
 {
 	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+#ifdef CONFIG_PM
 	struct uart_stm32_data *data = DEV_DATA(dev);
+#endif
+	int key;
 
 	/* Wait for TXE flag to be raised */
 	while (1) {
-		if (atomic_cas(&data->tx_lock, 0, 1)) {
+		if (LL_USART_IsActiveFlag_TXE(UartInstance)) {
```

Comment:
Please add a comment on what you're doing here since this is not obvious.
The commit message explain why previous solution was not working, 
but we also need to understand why current code is correct.
For this in line comments would be great

Reply:
I have added comments which hopefully clarify the code